routingpy/routers/graphhopper.py

```diff
@@ -380,7 +385,7 @@ def directions(  # noqa: C901
                     )
 
         return self._parse_directions_json(
-            self._request("/route", get_params=params, dry_run=dry_run), algorithm, elevation
+            self.client._request("/route", get_params=params, dry_run=dry_run), algorithm, elevation
```

Comment:
you see e.g. here: the `_request` method accepts a client_kwargs that is never passed. that was an oversight by us though: the `directions()` method also doesn't accept a client/requests_kwargs argument. 

The way this generally works is you have 3 hierarchies of "settings" via the kwargs:
1. global
2. per route class instance
3. per single request

While priority increases, i.e. single request > instance > global. A bit overloaded though..

So I propose: let's remove the `client_kwargs` from `_request` entirely and we only let options 1. and 2. active. That's more than enough anyways.

Hope that was clear.